### PR TITLE
added <fuel> support for SE|Hydrogen, biomass/coal split for SE|Solids

### PR DESCRIPTION
**Pull request by robertpietzcker**

### CODE REVIEW

**Review by danielhuppmann — APPROVED** [review]
Looks good, thanks @robertpietzcker

**Comment by robertpietzcker** [comment]
removed <fuel> implementation again, instead added the explicit gas/coal versions.
(was unsure about implications of the <fuel> thing, need more time to understand that

**Comment by robertpietzcker** [comment]
oh, I thought I could do a second (separate) pull request with the unit changes for the economy. sorry that this got automatically included in the first pull request.

File: definitions/variable/economy/economy.yaml [hunk 1]
```diff
@@ -2,69 +2,69 @@
 
 - Consumption:
     description: Total consumption of all goods, by all consumers in a region
-    unit: [million EUR_2020/yr, billion USD_2010/yr]
+    unit: [million EUR_2020/yr, billion USD_2010/yr, billion EUR_2020/yr]
```
Comment (danielhuppmann) [inline definitions/variable/economy/economy.yaml:5]: Let us please use only either million EUR or billion EUR - having the same unit at different orders of magnitude is basically an invitation for confusion...

File: definitions/variable/energy/energy-secondary.yaml [hunk 2]
```diff
@@ -56,6 +56,24 @@
 - Secondary Energy|Hydrogen|Fossil|w/o CCS:
     description: Hydrogen production from fossil fuels with freely vented CO2 emissions
     unit: EJ/yr
+- Secondary Energy|Hydrogen|Coal:
```
Comment (danielhuppmann) [inline definitions/variable/energy/energy-secondary.yaml:59]: Might as well use "Secondary Energy|Hydrogen|<Fuel>" and remove the per-fuel variables...

**Review by danielhuppmann — CHANGES_REQUESTED** [review]
Two comments inline.

File: definitions/variable/economy/economy.yaml (hunk 1 above)
Comment (robertpietzcker) [reply definitions/variable/economy/economy.yaml:5]: totally fine with this - I was afraid that if I remove something, I would interfere with the use of the nomenclature in another project. 
I take your comment as implying that I can remove the million EUR unit, so I will do a commit with this

File: definitions/variable/energy/energy-secondary.yaml (hunk 2 above)
Comment (robertpietzcker) [reply definitions/variable/energy/energy-secondary.yaml:59]: I have to admit I don't fully understand the comment?
or do you mean I should simply commit "Secondary Energy|Hydrogen|<Fuel>" and remove the explicit types?

Comment (danielhuppmann) [reply definitions/variable/energy/energy-secondary.yaml:59]: yes, please re-insert `Secondary Energy|Hydrogen|<Fuel>` and delete all explicit hydrogen-fuel variables (double-checking that the fuel is included in the Fuel list).

**Review by danielhuppmann — APPROVED** [review]
Thanks @robertpietzcker!

**Pull request merged**